common/persistence/execution_manager.go

```diff
@@ -300,6 +310,26 @@ func (m *executionManagerImpl) CreateWorkflowExecution(
 	return &CreateWorkflowExecutionResponse{HistorySize: newHistoryDiff}, nil
 }
 
+func (m *executionManagerImpl) serializeWorkflowEventBatches(
+	shardID int32,
+	eventBatches []*WorkflowEvents,
+) ([]*InternalAppendHistoryNodesRequest, int64, error) {
+	var workflowNewEvents []*InternalAppendHistoryNodesRequest
+	var historyDiff int64 = 0
+	for _, workflowEvents := range eventBatches {
+		newEvents, err := m.serializeWorkflowEvents(shardID, workflowEvents)
+		newEvents.ShardID = shardID
+		if err != nil {
+			return nil, 0, err
+		}
+		if newEvents != nil {
+			workflowNewEvents = append(workflowNewEvents, newEvents)
```

Comment:
when will newEvents be nil?

Reply:
can you make sure the newEvents is not nil?

from reading the code, newEvents is the serialized result of items within eventBatches, so if eventBatches does not contains nil element, this nil check is noop.

there should be no nil item in eventBatches in any condition (excluding bugs)
can you check why the nil check is necessary?